### ISSUE TITLE
How do you increase the text size in the TabViewItems header?

### ISSUE THREAD

**Issue by Going-Gone** [microsoft/microsoft-ui-xaml#3047]
I set
FontSize to a number, and the size stays the same. Is there something I'm missing?

`FontSize="500"`

**Comment by Felix-Dev** [comment]
@Going-Gone You can override the `TabViewItemHeaderFontSize` theme resource to set the FontSize for the header.

@StephenLPeters Should we template bind the TabViewItem's `FontSize` property here to its internal ContentPresenter?

**Comment by StephenLPeters** [comment]
Yup, I agree. We need to be better about remembering to bind these inherited properties. @ranjeshj

**Comment by StephenLPeters** [comment]
@MikeHillberg is FontSize Inherited? Should we need to template bind this to the template parts?

**Comment by StephenLPeters** [comment]
@Felix-Dev I think that the FontSize is intended to propagate to the tab's content, and the perscribed way of setting the header font size is through that light weight style.  If so, template binding the font size property to the header would be incorrect.

**Comment by MikeHillberg** [comment]
> @MikeHillberg is FontSize Inherited? Should we need to template bind this to the template parts?

FontSize inherits, but a lot of controls set it in their default style, which will override anything that comes from a parent.

**Comment by Felix-Dev** [comment]
>  I think that the FontSize is intended to propagate to the tab's content, and the perscribed way of setting the header font size is through that light weight style. If so, template binding the font size property to the header would be incorrect.

@StephenLPeters It does not have any effect at all on the Tab's content from what I can see:
```xml
<muxc:TabView>
    <muxc:TabView.TabItems>
        <muxc:TabViewItem Header="Tab 1" FontSize="80">
            Some tab content
        </muxc:TabViewItem>
    </muxc:TabView.TabItems>
</muxc:TabView>
```
![image](https://user-images.githubusercontent.com/1398851/89392218-8a1c2800-d709-11ea-9e72-e4637034dc03.png)

I also think if we have the `TabViewItem.FontWeight` property affecting the TabViewItem's header, so should `TabViewItem.FontSize`. Anything else would be unexpected API behavior to me. Alternatively, FontWeight. which is currently template bound, should be looked at again.

`TabViewItem.FontFamily` is inherited for the header:
```xml
<muxc:TabView x:Name="TabView" AddTabButtonClick="TabView_AddTabButtonClick"
              Grid.Row="1" >
    <muxc:TabView.TabItems>
        <muxc:TabViewItem Header="Tab 1" FontFamily="Segoe Script">
            Some tab content
        </muxc:TabViewItem>
    </muxc:TabView.TabItems>
</muxc:TabView>
```
![image](https://user-images.githubusercontent.com/1398851/89392971-79b87d00-d70a-11ea-9bc2-616fe2fdd5f0.png)

So from the way I see it - given the current API design - TabViewItem.FontSize should affect the font size of the tab's header - and only that.

**Comment by Going-Gone** [comment]
The TabView control has a FontSize property as well as the TabViewItem. Sort of confusing to what font it affect.

**Comment by StephenLPeters** [comment]
Hmm, @Felix-Dev your points are convincing, as the other API's are currently set up I agree I would expect font size to effect the header.  Unfortunate then that we already have a resouce which is driving this. Template binding the FontSize property would break anyone that was using the resource to customize this...

**Comment by StephenLPeters** [comment]
@Going-Gone I agree the scenario is confusing... @Felix-Dev if you set the FontSize property on the TabView does it do anything? Does setting the FontFamily on the TabView propagate to all of the TabViewItems?

**Comment by robloo** [comment]
I don't think we should be afraid of making breaking changes to fix template bindings. It just requires calling out clearly at the top of the release in a breaking changes section what apps need to update. This is especially true since upgrading WinUI is optional.

More generally, accumulation of errors is a problem that makes things tricky to use 10 years in the future. I brought this up in the past if we could setup a system were major releases were allowed breaking changes (like WinUI 4.0) where there is clear benefit (such as adopting ItemsRepeater across the board, etc.)

**Comment by StephenLPeters** [comment]
I agree this shouldn't block us fixing the issue. Just wanted to get the issue on the books.

**Comment by Felix-Dev** [comment]
@StephenLPeters The existing resource (TabViewItemHeaderFontSize), while not needed any longer if we template bind FontSize, would not cause a breaking change, if I add 
```
<Setter Property="FontSize" Value="{ThemeResource TabViewItemHeaderFontSize}"/>
```
 to the style of the TabViewItem.

Will check TabView.FontSize behavior soon.

**Comment by StephenLPeters** [comment]
I think that is true, but it would break the propigation behavior of FontSize, maybe that is okay though..

**Comment by Felix-Dev** [comment]
@StephenLPeters The TabView.FontSize is inherited by the ContentPresenter (TabContentPresenter) of the TabView:

```xml
<muxc:TabView FontSize="80" >
    <muxc:TabView.TabItems>
        <muxc:TabViewItem Header="Tab 1">
            Some tab content
        </muxc:TabViewItem>
        <muxc:TabViewItem Header="Tab 2"/>
    </muxc:TabView.TabItems>
</muxc:TabView>
```

![image](https://user-images.githubusercontent.com/1398851/89460813-32aea400-d76b-11ea-9f42-a790a5fdc361.png)

As for the propagation of TabViewItem.FontSize, I think it should be fine here because it is only used for the ContentPresenter of the TabViewItem's header. The TabViewItem Icon and Close button use their own values so we should be fine here with template binding TabViewItem's FontSize to the header ContentPresenter and setting it to the `TabViewItemHeaderFontSize` resource in the style. 

Or am I missing something here?

**Comment by StephenLPeters** [comment]
hmm, in that case these font properties feel a bit broken, I suspect we have this issue in a few places across the framework..

**Comment by ranjeshj** [comment]
@stmoy as FYI.

**Comment by stmoy** [comment]
Thanks for the CC - I'll add this to the small pile of "TabViewItem customization doesn't work quite right" issues 😉

**Comment by michael-hawker** [comment]
I actually had hit this just before this issue was filed, glad one was filed.

I believe the `FontSize` of the ContentPresenter being template bound could fix this issue, L690:

https://github.com/microsoft/microsoft-ui-xaml/blob/c6c8d084231f8cc4ffa978305e9f2a0f93a9a39c/dev/TabView/TabView.xaml#L682-L693

**Comment by hawkerm** [comment]
I just noticed a similar problem for the parent containers like TabView and ListView. If you set the Font properties on the parent ListView it has no effect on the child ListViewItems as I may expect...

**Comment by atiyka** [comment]
So couldn't anyone fix this issue in the last more than one year?
